vmtkScripts/contrib/vmtknumpytocenterlines.py

```diff
@@ -0,0 +1,111 @@
+##       University at Buffalo
+##       The Jacobs Institute
+
+from __future__ import absolute_import  # NEEDS TO STAY AS TOP LEVEL MODULE FOR Py2-3 COMPATIBILITY
+import vtk
+import sys
+
+from vmtk import vtkvmtk
+from vmtk import vmtkrenderer
+from vmtk import pypes
+
+try:
+    import numpy as np
+except ImportError:
+    raise ImportError('Unable to Import vmtknumpytosurface module, numpy is not installed')
+
+class vmtkNumpyToCenterlines(pypes.pypeScript):
+    def __init__(self):
+
+        pypes.pypeScript.__init__(self)
+
+        self.ArrayDict = None
+
+        self.SetScriptName('vmtkNumpyToCenterlines')
+        self.SetScriptDoc('Takes a nested python dictionary containg numpy arrays specifying vertex '
+                          'points, associated scalar data, and cell data specifying line connectivity'
+                          'and returns a VTK centerlines vtkPolyData file')
+        self.SetInputMembers([
+            ['ArrayDict', 'i', 'dict', 1, '', 'the input dictionary']])
+        self.SetOutputMembers([
+            ['Centerlines', 'o', 'vtkPolyData', 1, '', 'the output surface', 'vmtksurfacewriter']])
+
+    def Execute(self):
+
+        polyData = vtk.vtkPolyData()
+
+        points = vtk.vtkPoints()
+        for xyzPoint in self.ArrayDict['Points']:
+            points.InsertNextPoint(xyzPoint)
+
+        polyData.SetPoints(points)
+
+        self.PrintLog('converting numpy array to centerlines')
+        pointDataKeys = self.ArrayDict['PointData'].keys()
+        for key in pointDataKeys:
+
+            pointDataDType = str(self.ArrayDict['PointData'][key].dtype)
+            if pointDataDType.find('float') != -1:
+                pointDataArray = vtk.vtkFloatArray()
+            if pointDataDType.find('int') != -1:
+                pointDataArray = vtk.vtkIntArray()
+
+            try:
+                pointDataComponents = self.ArrayDict['PointData'][key].shape[1]
+            except IndexError:
+                pointDataComponents = 1
+
+            pointDataArray.SetNumberOfComponents(pointDataComponents)
+            pointDataArray.SetName(key)
+
+            if pointDataComponents == 1:
+                for pointData in self.ArrayDict['PointData'][key]:
+                    pointDataArray.InsertNextValue(pointData)
+                polyData.GetPointData().SetActiveScalars(key)
+                polyData.GetPointData().SetScalars(pointDataArray)
+            else:
+                for pointData in self.ArrayDict['PointData'][key]:
+                    pointDataArray.InsertNextTuple(pointData)
+                polyData.GetPointData().SetActiveVectors(key)
+                polyData.GetPointData().SetVectors(pointDataArray)
+
+        cellDataArray = vtk.vtkCellArray()
+        numberOfCells = len(self.ArrayDict['CellData']['CellPointIds'].keys())
+        for cellId in range(numberOfCells):
+            numberOfCellPoints = self.ArrayDict['CellData']['CellPointIds'][str(cellId)].size
```

Comment:
Hi Rick, I would suggest not to use a dictionary after 'CellPointIds', but a simple list, that you can index with an integer. You could use an integer as a key to the dictionary also, but there's no reason to do that, since `cellIds` are sequential and start from `0`. In any case, the `str(cellId)` doesn't really look good and I'd avoid it.

On a more general note, this layout (one entry per cell) is going to use more memory than needed for large datasets with many cells. It's not going to make a big difference for centerlines, but as a general strategy for other kinds of datasets it's going to be inefficient.

You could store cells more efficiently by using one or two arrays.
One-array solution: store cells sequentially in a NumPy array like 
```
cell0npts cell0pt0 cell0pt1 ... cell0pt4 cell1npts cell1pt0 cell1pt1 ...
```
This is quite compact, but it gets tricky if you every need to access a cell given its id.

Two-array solution: the above, plus an array with size `ncells` containing the index of each cell in the above array.

Last note: if you mix cell types (lines, polygons, etc) you might want to also have an array of size `ncells` containing the cell type for each cell.

It's your call, my only request is to get rid of the cellId as string thing. Thanks!